### ISSUE TITLE
Database error when upgrading 8.1.0 to 8.1.1 - also on 8.1.1 clean install

### ISSUE THREAD

**Issue by doominio** [PrestaShop/PrestaShop#33820]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I have a newly installed 8.1.0 (installed via Softaculous) that I'm testing and configuring. I noticed the available upgrade to 8.1.1, so I used 1-click upgrade.

This caused a database error that prevented the upgrade, this is the text:
`CRITICAL [console] Error thrown while running command "prestashop:schema:update-without-foreign --env=prod". Message: "An exception occurred while executing 'ALTER TABLE pshe_authorized_application CHANGE name name VARCHAR(50) NOT NULL':
`
I then tried a clean 8.1.1 install and got a few warnings and a database error as well:

```
[WARNING] Some commands could not be registered:
In Configuration.php line 331:
Argument 1 passed to PrestaShop\PrestaShop\Adapter\Configuration::restrictUpdatesTo() must be an instance of Shop, null given, called in /home/REDACTED/public_html/STORE/src/Core/Addon/Theme/ThemeManagerBuilder.php on line 61

[WARNING] Some commands could not be registered:
In Configuration.php line 331:
Argument 1 passed to PrestaShop\PrestaShop\Adapter\Configuration::restrictUpdatesTo() must be an instance of Shop, null given, called in /home/REDACTED/public_html/STORE/src/Core/Addon/Theme/ThemeManagerBuilder.php on line 61

CRITICAL [console] Error thrown while running command "prestashop:schema:update-without-foreign --env=prod". Message: "An exception occurred while executing 'ALTER TABLE ps_shop_url CHANGE id_shop_url id_shop_url INT AUTO_INCREMENT NOT NULL, CHANGE id_shop id_shop INT NOT NULL':

SQLSTATE[HY000]: General error: 1025 Error on rename of './DATABASENAME/ps_shop_url' to './DATABASENAME/#sql-backup-31ffdb-eb7131' (errno: 168 "Unknown (generic) error from engine")" ["exception" => Doctrine\DBAL\Exception\DriverException { …},"command" => "prestashop:schema:update-without-foreign --env=prod","message" => """ An exception occurred while executing 'ALTER TABLE ps_shop_url CHANGE id_shop_url id_shop_url INT AUTO_INCREMENT NOT NULL, CHANGE id_shop id_shop INT NOT NULL':\n \n SQLSTATE[HY000]: General error: 1025 Error on rename of './DATABASENAME/ps_shop_url' to './DATABASENAME/#sql-backup-31ffdb-eb7131' (errno: 168 "Unknown (generic) error from engine") """]
```

I'm running 10.6.15-MariaDB on a cPanel/Cloudlinux/Litespeed server.

### Steps to reproduce

1. Go to 1-click upgrade and follow the upgrade process. It will have the errors above.
2. Same with a clean install.

### PrestaShop version(s) where the bug happened

FROM 8.1.0 TO 8.1.1

### PHP version(s) where the bug happened

7.4.33

**Comment by hibatallahAouadni** [comment]
Hello @doominio 

I didn't manage to reproduce the issue with mysql DB, it seems the issue only occurs with MariaDB 😞 

[issue_33820.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/c8f41cbc-0fe3-4a64-964b-0816bd23365e)

Ping @PrestaShop/committers or @SD1982 can someone please try to reproduce the issue 🙏 
Waiting for your feedback.

Thanks!

**Comment by SD1982** [comment]
Hello @hibatallahAouadni and @doominio 
I can't totally reproduce the issue because I can't complete the installation process due to the same error noticed by @doominio 

![screencapture-localhost-8074-ps810-install-index-php-2023-09-05-11_45_47](https://github.com/PrestaShop/PrestaShop/assets/40635814/1afaa952-3099-4c1f-85f0-c8febd762cf6)

So there is definitely an error with mariadb ...
I tried with 2 version of mariadb (10.1 and 10.6)

**Comment by hibatallahAouadni** [comment]
Related to #33157 & #33337

**Comment by AureRita** [comment]
Hi @doominio 

Currently I tried to reproduce your issue (I know it's a bit too long) and currently I can't with all new features as you can see : 

[recording(270).webm](https://github.com/user-attachments/assets/572b2115-31a4-4ea0-9da4-09e5a296f3a4)

Currently, because I can't reproduce and the 7 september 2023, a PR that should fixed this issue seems to be done, I'll close this issue. 

If you have this issue again, please open a new issue, we will take care more quickly

Thank you

**Comment by AureRita** [comment]
fixed by #33874
